### ISSUE TITLE
multiple owlcarousel2 inside hidden (element) menu panel

### ISSUE THREAD

**Issue by claudchan** [OwlCarousel2/OwlCarousel2#597]
Hi,
I have an responsive advance menu and there will have multiple owl carousel will be init inside each sub panel. Here is my example jsfiddle:
http://jsfiddle.net/claudchan/eqbL6vLb/66/
First time load is good but I have problem when resizing the windows. Once if you resize it, you will notice the owl carousel is not properly adjusted. I trying hard to understand how to use owl carousel api events but I couldn't make my self to understand. I am using bootstrap collapse plugin. Appreciated if anyone could help. Thanks!

**Comment by joeyjoejoejr** [comment]
Can someone verify that this is still an issue?

**Comment by amaroktech** [comment]
I'm still having the issue. Using bootstrap tabs/pills.  One carousel inside of each tab.  The initial shown tab works fine, but the other tabs.. when switched to .. cause the thumbnails act just like this: http://stackoverflow.com/questions/26314797/owl-carousel-issue-with-bootstrap-tabs

**Comment by Hintalo** [comment]
yes, that is strange, the old version of OwlCarousel didn't had this
problem. In OwlCarousel2, per default invisible carousels do not work as
they are set to visible. Any chance that this will be fixed? Sadly, this
issue blocks some of us in upgrading to the new OwlCarousel2.

**Comment by karl-pogi** [comment]
I do use the OwlCarousel 2 inside Foundation5 tabs, and it indeed doesn't work, and the width gets messed up, but when I resize the browser window, it fixes itself.  I guess going back to OwlCarousel 1 would be the best option for now, but then lose that auto-width option that I like so much in v2.

**Comment by karl-pogi** [comment]
I managed to make it work btw.

added a click function then assigned and called for the carousel again.  The carousel now works properly from a hidden div container (tab).

$('#feat-tab2').click(function (e) {
    var owl2 = $('.owl-carousel2');
    owl2.owlCarousel({
        loop: false,
        margin: 20,
        nav: true,
        navRewind: false,
        dots: true,
        autoWidth: true,
        items: 2
    });
});

**Comment by dudewad** [comment]
I'm reproducing the same issue, and believe that this is a legitimate bug. I have actually written a workaround INTO the carousel itself, however it leverages the existing onResize() method rather than implementing it's own.
I added a new event type - _forceRefresh - which you can trigger on the owl carousel element like other events. That refresh is wired into the events proxy list and simply calls, as mentioned before, the onResize method.
While it's not a perfect implementation (ideally it should be calling a refresh() method), it works wonderfully. If I have time I will submit a pull request to the Owl team.

As a side node, @karl-pogi its great that you have a workaround from a calling block of code, however re-initializing the carousel hardly seems like a good approach (not that I need to tell you that) and ideally the component itself should handle recovery from something as simple as hide: resize window: show.

**Comment by jiMuBao** [comment]
save same issue, any other work around ?

**Comment by Kopakov** [comment]
I used Bootstrap collapse (hidden content). And I had the same bug. The reason was that class "owl-stage-outer" had "height: 0px". I gues owl javascript sets this height .

Solution in my case was:
.my-some-owl-carousel-class **.owl-stage-oute**r {
      **height: auto !important;**
}

**Comment by ikandee** [comment]
I too have this issue... hopefully this will be resolved.

**Comment by xd0** [comment]
I also have this issue, any news about this bug? @daviddeutsch

**Comment by cschweda** [comment]
Any fix yet?
